lib/linter.js

```diff
@@ -699,12 +744,23 @@ function runRules(sourceCode, configuredRules, ruleMapper, parserOptions, parser
 
     Object.keys(configuredRules).forEach(ruleId => {
         const severity = ConfigOps.getRuleSeverity(configuredRules[ruleId]);
+        const rule = ruleMapper(ruleId);
+
+        /*
+         * report an linting error, if the rule:
+         * a. has been replaced by some other rules;
+         * b. has not been defined;
+         */
+        if (!rule) {
+            lintingProblems.push(createMissingProblem(ruleId));
+            return;
+        }
```

Comment:
this should be moved to line 762, to avoid report non-existent rule in a config file. per https://github.com/eslint/eslint/issues/9505#issuecomment-491055516